coord/src/module.c

```diff
@@ -492,7 +492,7 @@ void prepareOptionalTopKCase(searchRequestCtx *req, RedisModuleString **argv, in
   if(queryNode!= NULL && queryNode->type == QN_VECTOR) {
     QueryVectorNode queryVectorNode = queryNode->vn;
     size_t k = queryVectorNode.vq->knn.k;
-    const char* scoreField = queryVectorNode.vq->scoreField;
+    const char* scoreField = queryNode->opts.distField ? queryNode->opts.distField : queryVectorNode.vq->scoreField;
```

Comment:
Why is this needed?

Reply:
If we have `queryNode->opts.distField` set, then user gave a distance field as a query attribute. Otherwise, either the user specified a distance field name in the `AS` clause (`[KNN ... AS v_dist]`), or we have the default field stored in `queryVectorNode.vq->scoreField`. As I explained [here](https://github.com/RediSearch/RediSearch/pull/3105#discussion_r977641769), if both fields are not NULL, we get an error anyway.

Reply:
We can have a `distField` in addition to the `scoreField`.
In the future, if we want to calculate something both, we will have it already.